presto-main/src/main/java/io/prestosql/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1278,14 +1278,26 @@ private void analyzeFiltersAndMasks(Table table, QualifiedObjectName name, Optio
                     .withRelationType(RelationId.anonymous(), new RelationType(fields))
                     .build();
 
+            Set<String> allColumns = fields.stream()
+                    .flatMap(field -> field.getName().stream())
+                    .collect(toImmutableSet());
+
+            Set<String> visibleColumns = accessControl.filterColumns(session.toSecurityContext(), name.asCatalogSchemaTableName(), allColumns);
```

Comment:
This assumes that filterColumns agrees with the "has select permissions" on those columns. I'm not sure that's guaranteed to be always true. In the worst case (if filterColumns is more restrictive), we might miss adding a mask over a field that has one.

Reply:
Identity can see the column if they has any privilege to that column. Typically select. But it could be insert too. So we don't collect masks only if identity is not allowed to see a columns, when they has no access to column.

Reply:
The case you are describing might cause that `DESCRIBE x` could return less columns than `SELECT * FROM x`, which does not make sense to me.

Reply:
There's nothing preventing that from happening, though. I think that's a deficiency in the access control APIs -- there should be a single method to get/check for permissions for columns.

Reply:
That is correct, the custom SPI implementation is out of our control. However we do require certain contract in many SPI parts. To me it is one place like that. The only problem here is that we don't have generic tests for access controls, where one could verify contract implementation. Such tests are impossible to write at the moment because, we can't simply "insert" authorization rules.

What do you suggest then? Previously I had called `checkCanSelectFromColumns` with one column at a time in a loop, but this was so ugly...

Reply:
I think it’s better to rely on chrckCanSelect. Someone may choose to use filterColumns as a way to obfuscate the available columns without preventing access to them, and the result would be a security bypass if there are column masks involved.